src/runtime-rs/crates/resource/src/cpu_mem/cpu.rs

```diff
@@ -0,0 +1,32 @@
+// Copyright (c) 2019-2022 Alibaba Cloud
+// Copyright (c) 2019-2022 Ant Group
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+use anyhow::{Context, Result};
+use kata_types::config::TomlConfig;
+use std::sync::Arc;
+
+#[derive(Default, Debug, Clone, Copy)]
+pub struct CpuResource {
+    /// The default vcpu number when booting
+    pub(crate) default_vcpu: i32,
+
+    /// Current number of vCPUs
+    pub(crate) current_vcpu: i32,
+}
+
+impl CpuResource {
+    pub fn new(config: Arc<TomlConfig>) -> Result<Self> {
+        let hypervisor_name = config.runtime.hypervisor_name.clone();
+        let hypervisor_config = config
+            .hypervisor
+            .get(&hypervisor_name)
+            .context("failed to get hypervisor")?;
+        Ok(Self {
+            default_vcpu: hypervisor_config.cpu_info.default_vcpus,
```

Comment:
I think you don't need a struct to store this value because it's easy to get it from the config. Also, I think you don't need the `CpuResource` struct, you can store the `current_vcpu` in the resource manager.